### PR TITLE
feat(connector): [Authorizedotnet] Add Wallet support

### PR DESCRIPTION
**Pull request by srujanchikke**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Wallet Support[Paypal, Googlepay and Applepay] for Authorizedotnet

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

**Note** : It takes one day for this connector to update status from Processing to Success.
Google pay 
Screenshot 2023-05-21 at 11 10 00 PM

Paypal Redirection 
Screenshot 2023-05-21 at 11 08 52 PM

Screenshot 2023-05-21 at 11 08 39 PM

Screenshot 2023-05-21 at 11 09 10 PM

tested unit tests to ensure existing flows
Screenshot 2023-05-21 at 11 15 58 PM

tested airwallex as there is core change which may alter airwallex 3ds payment 

Screenshot 2023-05-29 at 12 28 37 PM
Screenshot 2023-05-29 at 12 29 05 PM

**Note**  : Apple pay testing is pending

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (SanchithHegde) [inline crates/router/src/connector/authorizedotnet/transformers.rs:89]: Please remove this newline.

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (SanchithHegde) [inline crates/router/src/connector/authorizedotnet/transformers.rs:436]: Use `#[serde(rename_all = "PascalCase")]` instead.

**Review by SanchithHegde — APPROVED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:69]: why are other PaymentDetails removed? aren't they supported now?

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:80]: need both Serialize and Deserialize

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:87]: need both Serialize and Deserialize

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (jagan-jaya) [inline crates/router/src/connector/authorizedotnet/transformers.rs:94]: need both Serialize and Deserialize

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (srujanchikke) [reply crates/router/src/connector/authorizedotnet/transformers.rs:69]: Other payment methods are not implemented for this connector. They don't support this payment methods

File: crates/router/src/connector/authorizedotnet/transformers.rs
Comment (srujanchikke) [reply crates/router/src/connector/authorizedotnet/transformers.rs:87]: Parent struct uses both deserialize and serialize .

File: crates/router/src/core/payments.rs
Comment (ArjunKarthik) [inline crates/router/src/core/payments.rs:421]: Please get an approval from core team for this

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/build.rs
Comment (srujanchikke) [reply crates/router/src/build.rs]: i reverted this change.

File: crates/router/src/core/payments.rs
Comment (Narayanbhat166) [reply crates/router/src/core/payments.rs:421]: Yes, this is fine.

**Review by SanchithHegde — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**
